attestation-service/verifier/src/snp/mod.rs

```diff
@@ -158,18 +160,27 @@ pub(crate) fn verify_report_signature(
     vendor_certs: &VendorCertificates,
 ) -> Result<()> {
     // check cert chain
-    let VendorCertificates { ask, ark } = vendor_certs;
-    let vcek = verify_cert_chain(cert_chain, ask, ark)?;
+    let VendorCertificates { ask, ark, asvk } = vendor_certs;
+    let vcek = verify_cert_chain(cert_chain, ask, ark, asvk)?;
 
     // OpenSSL bindings do not expose custom extensions
     // Parse the vcek using x509_parser
     let vcek_der = &vcek.to_der()?;
     let parsed_vcek = X509Certificate::from_der(vcek_der)?.1.tbs_certificate;
 
-    // verify vcek fields
-    // chip id
-    if get_oid_octets::<64>(&parsed_vcek, HW_ID_OID)? != report.chip_id {
-        return Err(anyhow!("Chip ID mismatch"));
+    let subject = vcek.subject_name();
+    let common_name = subject
+        .entries_by_nid(Nid::COMMONNAME)
+        .next()
+        .and_then(|entry| entry.data().as_utf8().ok())
+        .map(|asn1| asn1.to_string());
+
+    if let Some(cn) = common_name {
+        if cn == "VCEK" {
+            if get_oid_octets::<64>(&parsed_vcek, HW_ID_OID)? != report.chip_id {
+                return Err(anyhow!("Chip ID mismatch"));
```

Comment:
We can always use `bail` to replace `return Err(anyhow!(...))`
```suggestion
                bail!("Chip ID mismatch");
```